src/forms/DateTimeLocalInput.jsx

```diff
@@ -71,7 +75,7 @@ DateTimeLocalInput.propTypes = {
 		PropTypes.string,
 		PropTypes.element,
 	]),
-	onChangeCallback: PropTypes.func
+	onChange: PropTypes.func, // provided by redux-form or by a wrapping component eg DateTimePicker
```

Comment:
since you're not setting the `value` based on internal `this.state` any more, I think `onChange` would be a `isRequired` property, and the parent _must_ update the `value` passed in as a `prop` in the onChange? If that's right, it would be good to document that implicit requirement - if the passed-in `onChange` prop _doesn't_ update `value`, then I think this component would lose its value every time it re-rendered